cpp/src/arrow/csv/parser_test.cc

```diff
@@ -360,6 +386,12 @@ TEST(BlockParser, Final) {
   csv = MakeCSVData({"ab,cd"});
   AssertParseFinal(parser, csv);
   AssertColumnsEq(parser, {{"ab"}, {"cd"}});
+
+  // Two blocks
+  auto csv1 = MakeCSVData({"ab,cd\n"});
+  auto csv2 = MakeCSVData({"ef,"});
+  AssertParseFinal(parser, {csv1, csv2});
```

Comment:
@pitrou I'm getting 'ambiguous function call' compilation errors with this and AssertParseOk(line 241). Not sure why this was not caught in CI builds. I am building with tests, gandiva, jni ON. Could you please take a look and let me know if I should set any compiler flags? Thanks.

Reply:
Can you post the full compiler output?

Reply:
`
/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:241:5: error: call to 'AssertParseOk' is ambiguous

    AssertParseOk(parser, {csv1, csv2});

    ^~~~~~~~~~~~~

/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:138:6: note: candidate function

void AssertParseOk(BlockParser& parser, const std::string& str) {

     ^

/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:144:6: note: candidate function

void AssertParseOk(BlockParser& parser, const std::vector<util::string_view>& data) {

     ^

/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:393:3: error: call to 'AssertParseFinal' is ambiguous

  AssertParseFinal(parser, {csv1, csv2});

  ^~~~~~~~~~~~~~~~

/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:150:6: note: candidate function

void AssertParseFinal(BlockParser& parser, const std::string& str) {

     ^

/Users/travis/build/dremio/arrow-build/arrow/cpp/src/arrow/csv/parser_test.cc:156:6: note: candidate function

void AssertParseFinal(BlockParser& parser, const std::vector<util::string_view>& data) {

     ^
`

Reply:
Can you try the following patch?
```diff
diff --git a/cpp/src/arrow/csv/parser_test.cc b/cpp/src/arrow/csv/parser_test.cc
index f988f3ce2..418340f54 100644
--- a/cpp/src/arrow/csv/parser_test.cc
+++ b/cpp/src/arrow/csv/parser_test.cc
@@ -135,16 +135,25 @@ Status ParseFinal(BlockParser& parser, const std::string& str, uint32_t* out_siz
   return parser.ParseFinal(util::string_view(str), out_size);
 }
 
+std::vector<util::string_view> ViewsFromStrings(const std::vector<std::string>& data) {
+  std::vector<util::string_view> views(data.size());
+  for (size_t i = 0; i < data.size(); ++i) {
+    views[i] = data[i];
+  }
+  return views;
+}
+
 void AssertParseOk(BlockParser& parser, const std::string& str) {
   uint32_t parsed_size = static_cast<uint32_t>(-1);
   ASSERT_OK(Parse(parser, str, &parsed_size));
   ASSERT_EQ(parsed_size, str.size());
 }
 
-void AssertParseOk(BlockParser& parser, const std::vector<util::string_view>& data) {
+void AssertParseOk(BlockParser& parser, const std::vector<std::string>& data) {
   uint32_t parsed_size = static_cast<uint32_t>(-1);
-  ASSERT_OK(parser.Parse(data, &parsed_size));
-  ASSERT_EQ(parsed_size, TotalViewLength(data));
+  auto views = ViewsFromStrings(data);
+  ASSERT_OK(parser.Parse(views, &parsed_size));
+  ASSERT_EQ(parsed_size, TotalViewLength(views));
 }
 
 void AssertParseFinal(BlockParser& parser, const std::string& str) {
@@ -153,10 +162,11 @@ void AssertParseFinal(BlockParser& parser, const std::string& str) {
   ASSERT_EQ(parsed_size, str.size());
 }
 
-void AssertParseFinal(BlockParser& parser, const std::vector<util::string_view>& data) {
+void AssertParseFinal(BlockParser& parser, const std::vector<std::string>& data) {
   uint32_t parsed_size = static_cast<uint32_t>(-1);
-  ASSERT_OK(parser.ParseFinal(data, &parsed_size));
-  ASSERT_EQ(parsed_size, TotalViewLength(data));
+  auto views = ViewsFromStrings(data);
+  ASSERT_OK(parser.ParseFinal(views, &parsed_size));
+  ASSERT_EQ(parsed_size, TotalViewLength(views));
 }
 
 void AssertParsePartial(BlockParser& parser, const std::string& str,
```

Reply:
same error with the patch

Reply:
enclosing elements of initializer list in braces seems to work. I pushed a patch - https://github.com/apache/arrow/pull/5791/files . Please merge if it looks alright. Thanks.

Reply:
Can you open a JIRA and a separate PR? This is needed in order to track the changes to Arrow.